linera-sdk/src/contract/test_runtime.rs

```diff
@@ -0,0 +1,652 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Runtime types to simulate interfacing with the host executing the contract.
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex, MutexGuard},
+};
+
+use linera_base::{
+    abi::ContractAbi,
+    data_types::{Amount, BlockHeight, Resources, SendMessageRequest, Timestamp},
+    identifiers::{Account, ApplicationId, ChainId, ChannelName, Destination, MessageId, Owner},
+    ownership::{ChainOwnership, CloseChainError},
+};
+use serde::Serialize;
+
+use crate::Contract;
+
+/// A mock of the common runtime to interface with the host executing the contract.
+pub struct ContractRuntime<Application>
```

Comment:
Should this be called a `MockContractRuntime`?

Reply:
Could we have both `mod runtime` and `mod test_runtime` declared in `mod.rs` (with appropriate flags)? Then, either `pub use self::runtime::ContractRuntime;` or `pub use self::test_runtime::MockContractRuntime as ContractRuntime;` depending on the compilation.

Reply:
Yes, good idea! Done.